RNA-seq/03-gastric_cancer_exploratory.Rmd

```diff
@@ -67,61 +67,29 @@ pca_plot_file <- file.path(plots_dir, "gastric-cancer_PC_scatter.pdf")
 
 ## DESeq2
 
-### Creating a DESeq2 dataset from tximport object
+### Creating a DESeq2 dataset from tximeta object
 
-First, let's read in the data we processed with `tximport`.
+First, let's read in the data we processed with `tximeta`.
 
 ```{r read-rds, live = TRUE}
 # Read in the RDS file we created in the last notebook
-txi <- readr::read_rds(txi_file)
-```
-
-```{r txi-counts, live = TRUE}
-# Let's look at the first few rows of the gene-level counts
-head(txi$counts)  
+gene_summarized <- readr::read_rds(txi_file)
 ```
 
-```{r txi-abundance, live = TRUE}
-# Let's look at the first few rows of the gene-level TPM
-head(txi$abundance) 
-```
-
-And now we'll read in the sample metadata.
-
-```{r sample_meta_df, live = TRUE}
-# Read in the sample metadata TSV file and have a look
-sample_meta_df <- readr::read_tsv(meta_file)
-sample_meta_df
-```
+### Set up DESeq2 object
 
-We'll save the column names of the counts to a variable that tells us the order of the sample names in `txi`.
+DESeq2 wants data to be rounded. 
 
-```{r col-names-txi}
-tximport_sample_names <- colnames(txi$counts)
-tximport_sample_names
-```
-
-We'll reorder the sample metadata to match the order in `txi`
-
-```{r arrange-sample_meta_df}
-sample_meta_df <- sample_meta_df %>%
-  dplyr::arrange(srr_accession)
-sample_meta_df
-```
-
-Check if these are in the same order.
-
-```{r all-equal}
-all.equal(tximport_sample_names, sample_meta_df$srr_accession)
+```{r}
+assay(gene_summarized) <- round(assay(gene_summarized))
```

Comment:
This feels a tad clumsy but 🤷‍♀️

Reply:
This isn't necessary any more. DESeq2 handles unrounded data just fine.

Reply:
DESeq2 complained which is why I added this step. Will try again though.